metricsbp/baseplate_hooks.go

```diff
@@ -101,3 +106,76 @@ var (
 	_ tracing.CreateChildSpanHook  = (*spanHook)(nil)
 	_ tracing.AddSpanCounterHook   = (*spanHook)(nil)
 )
+
+// NewConcurrencyCreateSpanHook returns a CreateServerSpanHook that can be used
+// to monitor the number of concurrent requests being processed and spawns a
+// reporter gorouting to publish that as a gauge.
+//
+// The reporter will update the gauge at the given interval.
+// The reporter goroutine will be stopped when the metrics context is canceled.
+// The gauge will be named "runtime.{hostname}.PID{process_id}.active_requests".
+func NewConcurrencyCreateSpanHook(metrics *Statsd, interval time.Duration) tracing.CreateServerSpanHook {
+	var activeRequests int64
+	counter := &atomicCounter{count: &activeRequests}
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = "UNKOWN-HOSTNAME"
+	}
+	gauge := metrics.Gauge(
+		fmt.Sprintf("runtime.%s.PID%d.active_requests", hostname, os.Getpid()),
```

Comment:
shouldn't the name be `concurrency` instead of `active_requests`?

Reply:
No it's `active_requests`, `concurrency` was a mistake.